pkgs/development/tools/boost-build/default.nix

```diff
@@ -32,8 +50,13 @@ stdenv.mkDerivation rec {
 
   installPhase = ''
     runHook preInstall
+
     ./b2 install --prefix="$out"
-    ln -s b2 "$out/bin/bjam"
+
+    # older versions of b2 created this symlink,
+    # which we want to support building via useBoost.
+    test -e "$out/bin/bjam" || ln -s b2 "$out/bin/bjam"
```

Comment:
```suggestion
    [ ! -e "$out/bin/bjam" ] && ln -s b2 "$out/bin/bjam"
```

Reply:
Reasoning?

Reply:
test is uncommon to be used and this makes the condition clearer.

Reply:
I prefer the `||` though.

I also use `test` which fits more natural into the language. The `]` is nonsense.

Reply:
whatever, UNRELATED